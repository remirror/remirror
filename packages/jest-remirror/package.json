{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
