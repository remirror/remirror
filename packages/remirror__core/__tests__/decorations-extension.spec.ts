import { extensionValidityTest, renderEditor } from 'jest-remirror';

import { AnyExtension, DecorationsExtension } from '../';

extensionValidityTest(DecorationsExtension);

describe('options', () => {
  describe('persistentSelectionClass', () => {
    it.each([
      {},
      { persistentSelectionClass: undefined },
      { persistentSelectionClass: '' },
      { persistentSelectionClass: 'selection' },
    ])('uses provided configuration (%j)', (configuration) => {
      const {
        add,
        dom,
        selectText,
        nodes: { p, doc },
      } = renderEditor<AnyExtension>([], { builtin: configuration });

      add(doc(p('Some text')));
      selectText({ from: 4, to: 9 });

      expect(dom.outerHTML).toMatchSnapshot();
    });

    it('retains selection and decoration', () => {
      const builtin = { persistentSelectionClass: 'selection' };
      const editor = renderEditor<never>([], { builtin });
      const {
        add,
        dom,
        selectText,
        nodes: { p, doc },
      } = editor;

      add(doc(p('Some text')));
      const range = { from: 4, to: 9 };
      selectText(range);

      // Verify that the selection has been applied
      expect(editor.state.selection.from).toBe(range.from);
      expect(editor.state.selection.to).toBe(range.to);

      (editor.view.dom as HTMLElement).blur();

      expect(dom.outerHTML).toMatchSnapshot();
      expect(editor.state.selection.from).toBe(range.from);
      expect(editor.state.selection.to).toBe(range.to);
    });
  });
});
