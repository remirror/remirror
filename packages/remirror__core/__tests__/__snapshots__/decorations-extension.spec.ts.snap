// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options persistentSelectionClass retains selection and decoration 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    Som
    <span class="selection">
      e tex
    </span>
    t
  </p>
</div>
`;

exports[`options persistentSelectionClass uses provided configuration ({"persistentSelectionClass":""}) 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    Some text
  </p>
</div>
`;

exports[`options persistentSelectionClass uses provided configuration ({"persistentSelectionClass":"selection"}) 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    Som
    <span class="selection">
      e tex
    </span>
    t
  </p>
</div>
`;

exports[`options persistentSelectionClass uses provided configuration ({}) 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    Some text
  </p>
</div>
`;

exports[`options persistentSelectionClass uses provided configuration ({}) 2`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    Some text
  </p>
</div>
`;
